students/DanCornutt/session08/circle.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+
+"""Circle HW, Session 07
+
+"""
+from math import pi
+
+class Circle:
+    """class for Circles with attributes and methods used by all."""
+    name = "Circle"
+    def __init__(self, radius):
+        self.radius = radius
+
+    @property
+    def diameter(self):
+        """Returns diameter of circle."""
+        return self.radius * 2
+
+    @diameter.setter
+    def diameter(self, value):
+        """Sets diameter of constructed circle.
+        param1: diameter of wanted circle"""
+        self.radius = value / 2
+
+    @property
+    def area(self):
+        """Computes area of circle. Returns area"""
+        return pi * self.radius ** 2
+
+    @classmethod
+    def from_diameter(cls, diameter):
+        """Alternate constructer for from_diameter
+        param1 : diameter of wanted circle"""
+        return cls(diameter / 2)
+
+
+    def __str__(self):
+        return "{} with radius {:.6f}".format(self.name, self.radius)
+
+
+    def __repr__(self):
+        return "{} with radius {:.6f}".format(self.name, self.radius)
+
+
+    def __add__(self, other):
+        try:
+            new = Circle(self.radius + other.radius)
+        except AttributeError:
+            new = Circle(self.radius + other)
+        return str(new)
+
+
+    def __radd__(self, other):
+        new = Circle(self.radius + other)
```

Comment:
not that freidnly to subclassing -- if you add two Spheres, you'll get a Circle :-)

you can use `self.__class__` instead.

Again, didn't really expect you to figure that out.